docs/serving/model-api.md

```diff
@@ -0,0 +1,239 @@
+can connect in a graph to form complex real-time pipelines.
+
+The basic serving function has a logical `router` with routes to multiple child `models`, 
+the url or the message will determine which model is selected, e.g. using the url schema:
+
+    /v2/models/<model>[/versions/<ver>]/operation
+
+> Note: the `model`, `version` and `operation` can also be specified in the message body 
+to support streaming protocols (e.g. Kafka).
+
+More complex routers can be used to support ensembles (send the request to all child models 
+and aggregate the result), multi-armed-bandit, etc. 
+
+You can use a pre-defined Router class, or write your own custom router. 
+Router can route to models on the same function or access models on a separate function.
+
+to specify the topology, router class and class args use `.set_topology()` with your function.
+
+### Creating Model Serving Function (Service)
+
+In order to provision a serving function we need to create an MLRun function of type `serving`
+, this can be done by using the `code_to_function()` call from a notebook. We can also import 
+an existing serving function/template from the marketplace.
+
+Example (run inside a notebook): this code converts a notebook to a serving function and adding a model to it:
+
+```python
+from mlrun import code_to_function
+fn = code_to_function('my-function', kind='serving')
+fn.add_model('m1', model_path=<model-artifact/dir>, class_name='MyClass', x=100)
+``` 
+
+see `.add_model()` docstring for help and parameters
+
+> See the full [Model Server example](https://github.com/mlrun/functions/blob/master/v2_model_server/v2_model_server.ipynb).
+
+If we want to use multiple versions for the same model, we use `:` to seperate the name from the version, 
+e.g. if the name is `mymodel:v2` it means model name `mymodel` version `v2`.
+
+User should specify the `model_path` (url of the model artifact/dir) and the `class_name` name 
+(or class `module.submodule.class`), alternatively you can set the `model_url` for calling a 
+model which is served by another function (can be used for ensembles).
+
+the function object(fn) accepts many options, you can specify replicas range (auto-scaling), cpu/gpu/mem resources, add shared 
+volume mounts, secrets, and any other Kubernetes resource through the `fn.spec` object or fn methods.
+
+e.g. `fn.gpu(1)` means each replica will use one GPU.
+
+to deploy a model we can simply call:
+
+```python
+fn.deploy()
+```
+
+we can also deploy a model from within an ML pipeline (check the various demos for details).
+
+## Model Server API
+
+MLRun Serving follows the same REST API defined by Triton and [KFServing v2](https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md)
+
+Nuclio also support streaming protocols (Kafka, kinesis, MQTT, ..), in the case of streaming 
+the `model` name and `operation` can be encoded inside the message body
+
+### get server info
+
+    GET / or /v2/health
```

Comment:
Not clear, I assume you meant there are 2 options

Reply:
right u can do either

Reply:
splitted to two lines